src/Drawer/Drawer.js

```diff
@@ -92,34 +87,244 @@ class Drawer extends React.Component {
     firstMount: true,
   };
 
-  componentWillReceiveProps() {
+  componentDidMount() {
+    if (this.props.variant === 'temporary') {
+      this.enableSwipeHandling();
+    }
+    this.mounted = true;
+  }
+
+  componentWillReceiveProps(nextProps) {
     this.setState({
       firstMount: false,
     });
+
+    if (this.props.variant !== 'temporary' && nextProps.variant === 'temporary') {
+      this.enableSwipeHandling();
+    } else if (this.props.variant === 'temporary' && nextProps.variant !== 'temporary') {
+      this.disableSwipeHandling();
+    }
+  }
+
+  componentWillUnmount() {
+    this.disableSwipeHandling();
+    this.removeBodyTouchListeners();
+    this.mounted = false;
+  }
+
+  onBodyTouchStart = event => {
+    // sometimes the event handler is called after unbinding it
+    if (!this.mounted) return;
+
+    const anchor = this.getAnchor();
+    const swipeAreaWidth = this.props.swipeAreaWidth;
+
+    const touchStartX =
+      anchor === 'right'
+        ? document.body.offsetWidth - event.touches[0].pageX
+        : event.touches[0].pageX;
+    const touchStartY =
+      anchor === 'bottom'
+        ? window.innerHeight - event.touches[0].clientY
+        : event.touches[0].clientY;
+
+    if (!this.props.open) {
+      if (this.isHorizontalSwiping()) {
+        if (touchStartX > swipeAreaWidth) return;
+      } else if (touchStartY > swipeAreaWidth) return;
+
+      if (this.props.disableSwipeToOpen) return;
+    }
+
+    this.maybeSwiping = true;
+    this.forceUpdate();
```

Comment:
Instead of this brute force method, what about using a state property?

Reply:
You're absolutely right! `maybeSwiping` is even needed for rendering, I'll move it into the state.